Add cerebrum

DewSecOff opened the pull request:
Add Cerebrum plugin to the Caido store

# I am submitting a new Plugin Package

## Repository URL

https://github.com/DewSecOff/Caido-Plugin-Cerebrum

Link to my plugin package: https://github.com/DewSecOff/Caido-Plugin-Cerebrum/releases/tag/1.0.1

## Release Checklist

- [ ] I have tested the plugin on
  - [ ] Windows
  - [ ] macOS
  - [x] Linux
- [x] My GitHub release contains all required files
  - [x] `plugin_package.zip`
  - [x] `plugin_package.zip.sig`
- [x] GitHub Tag name matches the exact version number specified in my `manifest.json`
- [x] The `id` in my `manifest.json` matches the `id` in the `plugin_packages.json` file.
- [x] My `README.md` describes the plugin package purpose and provides clear usage instructions.
- [x] I have read the developer policy at <https://developer.caido.io/policy.html>, and have assessed my plugin package adherence to this policy.
- [x] I have added a license in the `LICENSE` file and it matches the `license` field in the `plugin_packages.json` file.
- [x] My project respects and is compatible with the original license of any third-party code that I'm using.
      I have given proper attribution to these other projects in my `README.md` and/or `LICENSE`.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/caido/store?pullRequest=33) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/caido/store?pullRequest=33) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/caido/store?pullRequest=33) it.</sub>

Sytten commented:
Analysis:
- You probably dont want to run the dump on every saved request: https://github.com/DewSecOff/Caido-Plugin-Cerebrum/blob/bf5082ed51a289710b9d659be6c70f4ac8dae6de/packages/backend/src/index.ts#L18
- You could only run https://github.com/DewSecOff/Caido-Plugin-Cerebrum/blob/bf5082ed51a289710b9d659be6c70f4ac8dae6de/packages/backend/src/index.ts#L79-L94 on plugin init, no need to run it on every saved request, also do remember that if you need to add fields in the future you will need to do a database migrations since the table will already exist
- You should be able to do `import { Database } from 'sqlite'` instead of https://github.com/DewSecOff/Caido-Plugin-Cerebrum/blob/bf5082ed51a289710b9d659be6c70f4ac8dae6de/packages/backend/src/index.ts#L4 also you dont seem to use that variable anywhere

The frontend is good enough, just improve the small points above and we will get it merged :)

DewSecOff commented:
Hi @Sytten ! Thank you for your advices. I just modified the code.

- Moved table creation from `saveRequest` to plugin `init()` to avoid unnecessary execution
- Removed unused libs, function `dumpAllRequests` and others log call
- Added `.gitignore` to exclude `dist/`, `node_modules/`, and build artifacts

Sytten commented:
Hey! I was on vacation, just got back and will look at it this week @darklotuskdb

DewSecOff commented:
Thank you ! I try to unlock the CPTS currently, but I will add some features when I finish it :)
